llvm/lib/ObjectYAML/DXContainerEmitter.cpp

```diff
@@ -275,11 +275,14 @@ void DXContainerWriter::writeParts(raw_ostream &OS) {
 
       for (DXContainerYAML::RootParameterLocationYaml &L :
            P.RootSignature->Parameters.Locations) {
-        dxbc::RTS0::v1::RootParameterHeader Header{L.Header.Type, L.Header.Visibility,
-                                         L.Header.Offset};
 
-        switch (L.Header.Type) {
-        case llvm::to_underlying(dxbc::RootParameterType::Constants32Bit): {
+        mcdxbc::RootParameterHeader Header{
+            static_cast<dxbc::RootParameterType>(L.Header.Type),
+            static_cast<dxbc::ShaderVisibility>(L.Header.Visibility),
+            L.Header.Offset};
+
+        switch (Header.ParameterType) {
```

Comment:
Note that the `default` label in this switch becomes unreachable when we're switching over the enum itself, so it will need to be removed.